File: integrationtest/src/main/java/com/sequenceiq/it/SuiteContext.java
package com.sequenceiq.it;

import java.util.Collections;
import java.util.HashMap;
import java.util.Map;

public class SuiteContext {
    private Map<String, IntegrationTestContext> suiteContextMap = Collections.synchronizedMap(new HashMap<String, IntegrationTestContext>());

    public IntegrationTestContext getItContext(String suite) {
        return suiteContextMap.get(suite);
    }

    public IntegrationTestContext putItContext(String suite, IntegrationTestContext itContext) {
        return suiteContextMap.put(suite, itContext);
    }
}

File: integration-test/src/main/java/com/sequenceiq/it/SuiteContext.kt
package com.sequenceiq.it

import java.util.Collections
import java.util.HashMap

class SuiteContext {
    private val suiteContextMap = Collections.synchronizedMap(HashMap<String, IntegrationTestContext>())

    fun getItContext(suite: String): IntegrationTestContext {
        return suiteContextMap[suite]
    }

    fun putItContext(suite: String, itContext: IntegrationTestContext): IntegrationTestContext {
        return suiteContextMap.put(suite, itContext)
    }
}
